src/util/segmentation/operations/eraseInsideCircle.js

```diff
@@ -0,0 +1,58 @@
+import { getBoundingBoxAroundCircle } from '../boundaries';
+import { pointInEllipse } from '../../ellipse';
+import getCircleCoords from '../../getCircleCoords';
+import { getLogger } from '../../logger.js';
+
+const logger = getLogger('util:segmentation:operations:eraseInsideCircle');
+
+/**
+ * EraseInsideCircle - Erase all pixels labeled with the activeSegmentIndex,
+ * in the region defined by the circle.
+ * @param  {} evt The Cornerstone event.
+ * @param {} evt.operationData An object containing the `pixelData` to
+ *                          modify, the `segmentIndex` and the `points` array.
+ * @returns {null}
+ */
+export default function eraseInsideCircle(evt) {
+  const eventData = evt.detail;
+  const { operationData } = evt;
+
+  if (operationData.segmentationMixinType !== `circleSegmentationMixin`) {
+    logger.error(
+      `eraseInsideCircle operation requires circleSegmentationMixin operationData, recieved ${
+        operationData.segmentationMixinType
+      }`
+    );
+
+    return;
+  }
+
+  const { pixelData, segmentIndex } = operationData;
+
+  const { image } = eventData;
+  const { width } = image;
+  const [topLeft, bottomRight] = getBoundingBoxAroundCircle(evt);
+  const [xMin, yMin] = topLeft;
+  const [xMax, yMax] = bottomRight;
+  const ellipse = getCircleCoords(
+    evt.detail.handles.start,
+    evt.detail.handles.end
+  );
+
+  for (let x = xMin; x < xMax; x++) {
+    for (let y = yMin; y < yMax; y++) {
+      const inside = pointInEllipse(ellipse, {
+        x,
+        y,
+      });
+
+      if (inside) {
```

Comment:
Similar thought here as polygon erase. Is it worth short-circuiting on segment index match before checking `pointInEllipse`?

Reply:
To consolidate all of these `eraseInsideX.js` functions, could we just add an option for the `isPointInSomeShape`? I _think_ that's the only difference between all of them; with `rect`'s version basically being a function that always returns true.

Reply:
> Similar thought here as polygon erase. Is it worth short-circuiting on segment index match before checking pointInEllipse?

Yeah I'll do this for all of them.

> To consolidate all of these eraseInsideX.js functions, could we just add an option for the isPointInSomeShape? I think that's the only difference between all of them; with rect's version basically being a function that always returns true.

Yeah sure, we'll still need light wrapper functions as they are strategies though.

Reply:
We do this, kind of, for tool modes. We have `setToolMode`, but we created named functions from it using `bind` and then export them.

Reply:
TODO: Done most of this, few more to do